x/reward/simulation/decoder.go

```diff
@@ -0,0 +1,29 @@
+package simulation
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/provenance-io/provenance/x/reward/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types/kv"
+)
+
+// NewDecodeStore returns a decoder function closure that unmarshalls the KVPair's
+// Value
+func NewDecodeStore(cdc codec.Codec) func(kvA, kvB kv.Pair) string {
+	return func(kvA, kvB kv.Pair) string {
+		switch {
+		case bytes.Equal(kvA.Key[:1], types.RewardProgramIDKey):
+			var attribA, attribB types.RewardProgram
+
+			cdc.MustUnmarshal(kvA.Value, &attribA)
+			cdc.MustUnmarshal(kvB.Value, &attribB)
+
+			return fmt.Sprintf("%v\n%v", attribA, attribB)
```

Comment:
Don't we need a `case` for each type of key we put in the store?

Also, you can use `bytes.HasPrefix(kva.Key, types.RewardProgramIDKey)` to be a bit more robust (e.g. if `RewardProgramIDKey` ever changes to have more than one byte).

Reply:
i am not sure 
very specific to to sim tests for sure ``````// AppModuleSimulation defines the standard functions that every module should expose
// for the SDK blockchain simulator``````
`metadata` and `marker` both have it commented out ```// RegisterStoreDecoder registers a decoder for metadata module's types
func (am AppModule) RegisterStoreDecoder(sdr sdk.StoreDecoderRegistry) {
	// sdr[types.StoreKey] = simulation.NewDecodeStore(am.cdc)
}
```
 marker has a lot of sims tests, so not sure how it is used, but i added the rest of the keys(actually there were a couple of unused keys, so a helpful comment :) )